From 03bf8eef42212333c145577b7ef226affdae30c6 Mon Sep 17 00:00:00 2001
From: Dylan Hung <dylan_hung@aspeedtech.com>
Date: Fri, 15 Jul 2022 17:18:16 +0800
Subject: [PATCH 11/30] i3c: add SETHID CCC support

Add SETHID CCC support, this CCC shall be sent before SETAASA as
specified in JESD403 standard.

Signed-off-by: Dylan Hung <dylan_hung@aspeedtech.com>
Change-Id: I439c7c7aa2ec27c98990724ce1f9d8861f7ca79f
---
 drivers/i3c/i3c_common.c  | 15 +++++++++++++++
 include/drivers/i3c/i3c.h |  2 ++
 2 files changed, 17 insertions(+)

diff --git a/drivers/i3c/i3c_common.c b/drivers/i3c/i3c_common.c
index 261c4de5ea..400a572f1e 100644
--- a/drivers/i3c/i3c_common.c
+++ b/drivers/i3c/i3c_common.c
@@ -54,6 +54,21 @@ int i3c_master_send_rstdaa(const struct device *master)
 	return i3c_master_send_ccc(master, &ccc);
 }
 
+int i3c_master_send_sethid(const struct device *master)
+{
+	struct i3c_ccc_cmd ccc;
+	uint8_t hid = 0;
+
+	ccc.addr = I3C_BROADCAST_ADDR;
+	ccc.id = I3C_CCC_SETHID;
+	ccc.payload.length = 1;
+	ccc.payload.data = &hid;
+	ccc.rnw = 0;
+	ccc.ret = 0;
+
+	return i3c_master_send_ccc(master, &ccc);
+}
+
 int i3c_master_send_aasa(const struct device *master)
 {
 	struct i3c_ccc_cmd ccc;
diff --git a/include/drivers/i3c/i3c.h b/include/drivers/i3c/i3c.h
index 084a69f695..48b4634bda 100644
--- a/include/drivers/i3c/i3c.h
+++ b/include/drivers/i3c/i3c.h
@@ -21,6 +21,7 @@
 /* broadcast only commands */
 #define I3C_CCC_ENTDAA          0x07
 #define I3C_CCC_SETAASA         0x29
+#define I3C_CCC_SETHID		0x61
 
 /* unicast only commands */
 #define I3C_CCC_SETDASA         (0x7 | I3C_CCC_DIRECT)
@@ -220,6 +221,7 @@ int i3c_aspeed_slave_wait_data_consume(const struct device *dev);
 int i3c_master_send_enec(const struct device *master, uint8_t addr, uint8_t evt);
 int i3c_master_send_disec(const struct device *master, uint8_t addr, uint8_t evt);
 int i3c_master_send_rstdaa(const struct device *master);
+int i3c_master_send_sethid(const struct device *master);
 int i3c_master_send_aasa(const struct device *master);
 int i3c_master_send_setmrl(const struct device *master, uint8_t addr, uint16_t mrl,
 			   uint8_t ibi_payload_size);
-- 
2.24.1

